packages/Heading/src/Heading.js

```diff
@@ -53,55 +48,46 @@ const defaultProps = {
 
 const safeValue = n => (n < 1 || Number.isNaN(n) ? 6 : Math.min(n, 6));
 
-class Heading extends React.Component {
-  getElementProps(safeDisplayLevel, safeLevel) {
-    const {
-      a11yText,
-      children,
-      className,
-      displayLevel,
-      domRef,
-      hasDivider,
-      hasUnderline,
-      isHidden,
-      isLight,
-      isSemantic,
-      level,
-      ...moreProps
-    } = this.props;
-    return {
-      "aria-level": isSemantic ? null : safeLevel,
-      "aria-label": a11yText || undefined,
-      className: classNames(`heading--level-${safeDisplayLevel || safeLevel}`, className, {
-        "heading--is-hidden": isHidden,
-        "heading--has-underline": hasUnderline,
-        "heading--is-light": isLight,
-      }),
-      ref: domRef,
-      role: isSemantic ? null : "heading",
-      ...moreProps,
-    };
-  }
-
-  renderHeadingContent() {
-    const { a11yText, children } = this.props;
-    return a11yText ? <span aria-hidden>{children}</span> : children;
-  }
-
-  render() {
-    const { hasDivider, isSemantic, level, displayLevel } = this.props;
-    const safeLevel = safeValue(level);
-    const safeDisplayLevel = displayLevel ? safeValue(displayLevel) : null;
-    const divider = <span className="heading__divider" />;
-    const elementProps = this.getElementProps(safeDisplayLevel, safeLevel);
-
-    return (
-      <div css={headingStyles} {...elementProps} as={isSemantic ? `h${safeLevel}` : "div"}>
-        {this.renderHeadingContent()}
-        {hasDivider ? divider : null}
-      </div>
-    );
-  }
+function getElementProps(safeDisplayLevel, safeLevel, props) {
+  const {
+    a11yText,
+    children,
+    displayLevel,
+    domRef,
+    hasDivider,
+    hasUnderline,
+    isHidden,
+    isLight,
+    isSemantic,
+    level,
+    ...moreProps
+  } = props;
+  return {
+    "aria-level": isSemantic ? null : safeLevel,
+    "aria-label": a11yText || undefined,
+    ref: domRef,
+    role: isSemantic ? null : "heading",
+    ...moreProps,
+    ...props,
+  };
+}
+
+function renderHeadingContent(a11yText, children) {
+  return a11yText ? <span aria-hidden>{children}</span> : children;
+}
+
+function Heading(props) {
+  const { a11yText, children, hasDivider, isSemantic, level, displayLevel } = props;
+  const safeLevel = safeValue(level);
+  const safeDisplayLevel = displayLevel ? safeValue(displayLevel) : null;
+  const divider = <span className="heading__divider" />;
```

Comment:
You could export a style just for this <span>

For ex
```
<span css={dividerStyles} />
```

Then you export dividerStyles from the styles file. You can see some examples. We do this in other components

Reply:
done